pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -369,6 +381,116 @@ func (r *ReconcileComplianceRemediation) countKubeUnmetDependencies(rem *compv1a
+	annotations := rem.GetAnnotations()
+	labels := rem.GetLabels()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	requiredValues := annotations[compv1alpha1.RemediationValueRequiredAnnotation]
+	requiredValuesList := strings.Split(requiredValues, ",")
+	_, isRequiredValuesProcessed := labels[compv1alpha1.RemediationValueRequiredProcessedLabel]
+	var unsetRequireValueList []string
+	var setRequireValueList []string
+
+	//Go through values required list to find is it inside the tailored profile
+	for _, requiredValue := range requiredValuesList {
+		found, err := r.isRequiredValueSet(rem, requiredValue)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Error finding if required value is set: %v", err)
+		}
+		if found {
+			setRequireValueList = append(setRequireValueList, requiredValue)
+		} else {
+			unsetRequireValueList = append(unsetRequireValueList, requiredValue)
+		}
+
+	}
+
+	currentUnsetValues := strings.Split(annotations[compv1alpha1.RemediationUnsetValueAnnotation], ",")
+	currentUsedValues := strings.Split(annotations[compv1alpha1.RemediationValueUsedAnnotation], ",")
+	if !isRequiredValuesProcessed {
+		labels[compv1alpha1.RemediationValueRequiredProcessedLabel] = ""
+		if len(unsetRequireValueList) > 0 {
+			annotations[compv1alpha1.RemediationUnsetValueAnnotation] = strings.Join(append(currentUnsetValues, unsetRequireValueList...), ",")
+		}
+		if len(setRequireValueList) > 0 {
+			annotations[compv1alpha1.RemediationValueUsedAnnotation] = strings.Join(append(currentUsedValues, setRequireValueList...), ",")
+		}
+		rCopy := rem.DeepCopy()
+		rCopy.SetAnnotations(annotations)
+		rCopy.SetLabels(labels)
+		err := r.client.Update(context.TODO(), rCopy)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("adding xccdf required-value label/annotation: %w", err)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+//To find if it is set in tailored profile
+func (r *ReconcileComplianceRemediation) isRequiredValueSet(rem *compv1alpha1.ComplianceRemediation, requiredValue string) (bool, error) {
+	var scan = &compv1alpha1.ComplianceScan{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Namespace: rem.GetNamespace(),
+		Name:      rem.GetScan(),
+	}, scan)
+
+	if err != nil {
+		return false, err //error getting scan
+	}
+	if scan.Spec.TailoringConfigMap != nil {
+		tp := &compv1alpha1.TailoredProfile{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: scan.Spec.TailoringConfigMap.Name, Namespace: rem.GetNamespace()}, tp)
+		if err != nil {
+			return false, err
+		}
+		for _, setValue := range tp.Spec.SetValues {
+			if setValue.Name == requiredValue {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+	//scan did not have a tailored profile
+	return false, nil
+}
+
+func (r *ReconcileComplianceRemediation) handleUnsetValues(rem *compv1alpha1.ComplianceRemediation, logger logr.Logger) (reconcile.Result, error) {
```

Comment:
I'd like to see a unit test for this function, that would ensure this works as expected and help us reviewers understand better what this does.

Reply:
I will add a unit test for that!